### PR TITLE
[ENH, MRG] Add decimate to EpochsTFR object

### PR DESCRIPTION
**Pull request by alexrockhill**
Fixes memory failures here https://github.com/mne-tools/mne-python/pull/10920. 

I just needed the decimate method but while I was changing that, it made sense to add in `tmin` and `tmax` and to mak `times` readonly as in `Epochs`.

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
@drammock maybe merge if happy?

🙏 @alexrockhill !

File: mne/time_frequency/tfr.py [hunk 1]
```diff
@@ -2217,7 +2217,9 @@ def __init__(self, info, data, times, freqs, comment=None, method=None,
             (len(dl) == 0 for dl in drop_log))
         event_id = _check_event_id(event_id, events)
         self.data = data
-        self.times = np.array(times, dtype=float)
+        self._set_times(np.array(times, dtype=float))
+        self._raw_times = self.times.copy()
```
Comment (drammock) [inline mne/time_frequency/tfr.py:2221]:
```suggestion
        self._raw_times = self.times.copy()  # needed for decimate
```

**Review by drammock — APPROVED** [review]

**Comment by alexrockhill** [comment]
I didn't fix checking for aliasing which doesn't apply to `EpochsTFR`. It's in https://github.com/mne-tools/mne-python/pull/10920 but I can pull out and merge separately also as that's still a WIP.

**Comment by drammock** [comment]
> I didn't fix checking for aliasing which doesn't apply to `EpochsTFR`. It's in #10920 but I can pull out and merge separately also as that's still a WIP.

ah, dang. yes, please pull that out in a separate PR.

**Pull request merged**